CRM/Report/Form.php

```diff
@@ -4779,6 +4779,16 @@ protected function setOutputMode() {
       FALSE,
       CRM_Utils_Array::value('task', $this->_params)
     ));
+    if ($this->_outputMode !== NULL) {
+      // We didn't get it using output, so we now try with task.
+      $this->_outputMode = str_replace('report_instance.', '', CRM_Utils_Request::retrieve(
+          'task',
+          'String',
+          CRM_Core_DAO::$_nullObject,
+          FALSE,
+          CRM_Utils_Array::value('task', $this->_params)
+        ));
+    }
```

Comment:
I have just tried with our test system running 4.7.13 with a non-privileged logon, and the various tasks did nothing, but just returned you to the report.  Saving the report left the user with a blank report.  Switching to the custom code results in.
The Add Contact to Group option disappearing and the various actions working.  The Save action are no longer visible.